File: python-src/canny_edge.py
import numpy as np
import cv2

original = cv2.imread('sample_img/brick-house.png', 0)
threshold_val1 = 250
threshold_val2 = 300

canny = cv2.Canny(original, threshold_val1, threshold_val2)

cv2.imshow('canny', canny)
cv2.waitKey(0)

File: canny_edge.cpp
#include <stdio.h> 
#include "opencv2/core/core.hpp"
#include "opencv2/highgui/highgui.hpp"
#include <opencv2/imgproc/imgproc.hpp>

using namespace cv;

int main()
{        
	Mat image = cv::imread("sample_img/cat_01.jpg", CV_LOAD_IMAGE_COLOR);
	if (!image.data)
	{
		printf("Could not open or find the image\n");
		return -1;
	}		
	imshow("Original Image", image);

    Mat gray, draw, edge;
    cvtColor(image, gray, CV_BGR2GRAY);
    Canny( gray, edge, 100, 150, 3);
    edge.convertTo(draw, CV_8U);

    imshow("Canny Edge Detector", draw);

    waitKey(0);
	return 0;
}
